WebCryptoAPI/encrypt_decrypt/aes.js

```diff
@@ -0,0 +1,318 @@
+
+function run_test() {
+    var subtle = crypto.subtle; // Change to test prefixed implementations
+
+    // When are all these tests really done? When all the promises they use have resolved.
+    var all_promises = [];
+
+    // Source file aes_XXX_vectors.js provides the getTestVectors method
+    // for the AES-XXX algorithm that drives these tests.
+    var vectors = getTestVectors();
+    var passingVectors = vectors.passing;
+    var failingVectors = vectors.failing;
+    var decryptionFailingVectors = vectors.decryptionFailing;
+
+    // Check for successful encryption.
+    passingVectors.forEach(function(vector) {
+        var promise = importVectorKey(vector, ["encrypt", "decrypt"])
+        .then(function(vector) {
+            promise_test(function(test) {
+                return subtle.encrypt(vector.algorithm, vector.key, vector.plaintext)
+                .then(function(result) {
+                    assert_true(equalBuffers(result, vector.result), "Should return expected result");
+                }, function(err) {
+                    assert_unreached("encrypt error for test " + vector.name + ": " + err.message);
+                });
+            }, vector.name);
+        }, function(err) {
+            // We need a failed test if the importVectorKey operation fails, so
+            // we know we never tested encryption
+            promise_test(function(test) {
+                assert_unreached("importKey failed for " + vector.name);
+            }, "importKey step: " + vector.name);
+        });
+
+        all_promises.push(promise);
+    });
+
+    // Check for successful encryption even if the buffer is changed after calling encrypt.
+    passingVectors.forEach(function(vector) {
+        var plaintext = copyBuffer(vector.plaintext);
+        var promise = importVectorKey(vector, ["encrypt", "decrypt"])
+        .then(function(vector) {
+            promise_test(function(test) {
+                return subtle.encrypt(vector.algorithm, vector.key, plaintext)
+                .then(function(result) {
+                    plaintext[0] = 255 - plaintext[0];
+                    assert_true(equalBuffers(result, vector.result), "Should return expected result");
```

Comment:
Yeah, I just saw that when I was working on sign/verify. I'll get it fixed
and checked in first thing in the morning. I may be able to get sign and
verify done tomorrow, too.

Charlie

On Thu, Jun 9, 2016 at 6:28 PM, Jim Schaad <notifications@github.com> wrote:

> In WebCryptoAPI/encrypt_decrypt/aes.js
> <https://github.com/w3c/web-platform-tests/pull/3110#discussion_r66533404>
> :
>
> > +            }, "importKey step: " + vector.name);
> > +        });
> > +
> > +        all_promises.push(promise);
> > +    });
> > +
> > +    // Check for successful encryption even if the buffer is changed after calling encrypt.
> > +    passingVectors.forEach(function(vector) {
> > +        var plaintext = copyBuffer(vector.plaintext);
> > +        var promise = importVectorKey(vector, ["encrypt", "decrypt"])
> > +        .then(function(vector) {
> > +            promise_test(function(test) {
> > +                return subtle.encrypt(vector.algorithm, vector.key, plaintext)
> > +                .then(function(result) {
> > +                    plaintext[0] = 255 - plaintext[0];
> > +                    assert_true(equalBuffers(result, vector.result), "Should return expected result");
>
> This needs to be moved out of the encrypt promise.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/3110/files/bdc1e5774a682fb7baec2cc3db973af24b49ac97#r66533404>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAE_NyCxNlAUuy25KySKAq6zHRi2CMrsks5qKJOegaJpZM4ItxzV>
> .
>